components/icu/src/lib.rs

```diff
@@ -459,40 +459,72 @@ pub mod list {
     //!
     //! # Examples
     //!
-    //! ## Format a list of strings in Spanish
+    //! ## Formatting *and* lists in Spanish
     //!
     //! ```
-    //! use icu::list::{ListFormatter, ListStyle};
-    //! use icu::locid::locale;
-    //! use writeable::Writeable;
-    //!
-    //! let provider = icu_testdata::get_provider();
-    //! let list_formatter = ListFormatter::try_new_and(locale!("es"), &provider, ListStyle::Wide)
-    //!     .expect("Data should load successfully");
-    //!
-    //! assert_eq!(
-    //!     list_formatter
-    //!         .format(["España", "Suiza"].iter())
-    //!         .write_to_string(),
-    //!     "España y Suiza"
+    //! # use icu_list::{ListFormatter, ListStyle};
+    //! # use icu_locid::locale;
+    //! # use writeable::*;
+    //! #
+    //! let list_formatter = ListFormatter::try_new_and(
+    //!     locale!("es"),
+    //!     &icu_testdata::get_provider(),
+    //!     ListStyle::Wide,
+    //! )
+    //! .expect("Data should load successfully");
+    //!
+    //! assert_writeable_eq!(
+    //!     list_formatter.format(["España", "Suiza"].iter()),
+    //!     "España y Suiza",
     //! );
     //!
     //! // The Spanish 'y' sometimes becomes an 'e':
-    //! assert_eq!(
-    //!     list_formatter
-    //!         .format(["España", "Suiza", "Italia"].iter())
-    //!         .write_to_string(),
-    //!     "España, Suiza e Italia"
+    //! assert_writeable_eq!(
+    //!     list_formatter.format(["España", "Suiza", "Italia"].iter()),
+    //!     "España, Suiza e Italia",
     //! );
+    //! ```
     //!
-    //! // We can use any Writeables as inputs:
-    //! assert_eq!(
-    //!     list_formatter.format(1..=10).write_to_string(),
-    //!     "1, 2, 3, 4, 5, 6, 7, 8, 9 y 10"
+    //! ## Formatting *or* lists in Thai
+    //!
+    //! ```
+    //! # use icu_list::{ListFormatter, ListStyle};
+    //! # use icu_locid::locale;
+    //! # use writeable::*;
+    //! #
+    //! let list_formatter = ListFormatter::try_new_or(
+    //!     locale!("th"),
+    //!     &icu_testdata::get_provider(),
+    //!     ListStyle::Short,
+    //! )
+    //! .expect("Data should load successfully");
+    //!
+    //! // We can use any Writeables as inputs
+    //! assert_writeable_eq!(
+    //!     list_formatter.format(1..=3),
+    //!     "1, 2 หรือ 3",
     //! );
     //! ```
     //!
-    //! [`ListFormatter`]: ListFormatter
+    //! ## Formatting unit lists in English
+    //!
+    //! ```
+    //! # use icu_list::{ListFormatter, ListStyle};
+    //! # use icu_locid::locale;
+    //! # use writeable::*;
+    //! #
+    //! let list_formatter = ListFormatter::try_new_unit(
+    //!     locale!("en"),
+    //!     &icu_testdata::get_provider(),
+    //!     ListStyle::Wide,
+    //! )
+    //! .expect("Data should load successfully");
+    //!
+    //! assert_writeable_eq!(
+    //!     list_formatter.format(["1ft", "2in"].iter()),
```

Comment:
Nit: Add a comment that says you should visit https://github.com/unicode-org/icu4x/issues/2192 to track full-stack measurement unit formatting in ICU4X (since this example is not fully i18n-friendly).